pkg/rpcserver/rpcserver.go

```diff
@@ -171,43 +262,44 @@ func (serv *RPCServer) handleConn(conn *flatrpc.Conn) {
 	log.Logf(2, "runner %v: %v", name, err)
 }
 
-func (serv *RPCServer) handshake(conn *flatrpc.Conn) (string, []byte, *cover.CanonicalizerInstance, error) {
-	connectReqRaw, err := flatrpc.Recv[flatrpc.ConnectRequestRaw](conn)
+func (serv *Server) handshake(conn *flatrpc.Conn) (string, []byte, *cover.CanonicalizerInstance, error) {
+	connectReq, err := flatrpc.Recv[*flatrpc.ConnectRequestRaw](conn)
 	if err != nil {
 		return "", nil, nil, err
 	}
-	connectReq := connectReqRaw.UnPack()
-	log.Logf(1, "fuzzer %v connected", connectReq.Name)
+	log.Logf(1, "runner %v connected", connectReq.Name)
 	if !serv.cfg.VMLess {
 		checkRevisions(connectReq, serv.cfg.Target)
 	}
 	serv.statVMRestarts.Add(1)
 
-	bugFrames := serv.mgr.currentBugFrames()
+	leaks, races := serv.mgr.BugFrames()
 	connectReply := &flatrpc.ConnectReply{
-		Debug:      *flagDebug,
-		Procs:      int32(serv.cfg.Procs),
-		Slowdown:   int32(serv.cfg.Timeouts.Slowdown),
-		LeakFrames: bugFrames.memoryLeaks,
-		RaceFrames: bugFrames.dataRaces,
+		Debug:            serv.cfg.Debug,
+		Cover:            serv.cfg.Cover,
+		Procs:            int32(min(serv.cfg.Procs, prog.MaxPids)),
```

Comment:
In the code below, we are referring to `serv.cfg.Procs`. Should we then do the `min()` somewhere above and patch the `serv.cfg.Procs` field itself?

Reply:
Good point. Patching it in newImpl.